src/app.rs

```diff
@@ -217,19 +215,22 @@ impl App {
             .pending_identity_exists(group_id, &commitment)
             .await?
```

Comment:
Doesn't this have a race condition where two concurrent identical insert requests could both pass this point?

Out of scope for this PR, but let's open an issue for it.